guides/v2.3/inventory/source-selection-algorithms.md

```diff
@@ -0,0 +1,167 @@
+---
+group: inventory
+title: Source selection algorithms
+---
+
+The **Source Selection Algorithm (SSA)** receives the total number of shippable SKUs and their quantities in an order as well as the order's shipping address. It recommends how to fulfill partial and full shipments. The merchant decides which business needs take precedence when deciding which shipping method to use:
```

Comment:
>receives the total number of shippable SKUs and their quantities in an order as well as the order's shipping address. 

This statement is not really correct, in our current implementation, SSA accepts `InventoryRequestInterface` object.

```
namespace Magento\InventorySourceSelectionApi\Api;

/**
 * Returns source selection algorithm result for given Inventory Request
 *
 * @api
 */
interface SourceSelectionServiceInterface
{
    /**
     * @param \Magento\InventorySourceSelectionApi\Api\Data\InventoryRequestInterface $inventoryRequest
     * @param string $algorithmCode
     * @return \Magento\InventorySourceSelectionApi\Api\Data\SourceSelectionResultInterface
     */
    public function execute(
        \Magento\InventorySourceSelectionApi\Api\Data\InventoryRequestInterface $inventoryRequest,
        string $algorithmCode
    ): \Magento\InventorySourceSelectionApi\Api\Data\SourceSelectionResultInterface;
}

```

Which in its turn contains `Stock Id`  for which we want to apply calculation and list of requested items.

```
interface ItemRequestInterface extends \Magento\Framework\Api\ExtensibleDataInterface
{
    /**
     * Requested SKU
     *
     * @return string
     */
    public function getSku(): string;

    /**
     * Requested Product Quantity
     *
     * @return float
     */
    public function getQty(): float;

    /**
     * Set SKU
     *
     * @param string $sku
     * @return void
     */
    public function setSku(string $sku): void;

    /**
     * Set Quantity
     *
     * @param float $qty
     * @return void
     */
    public function setQty(float $qty): void;

    /**
     * Retrieve existing extension attributes object
     *
     * @return \Magento\InventorySourceSelectionApi\Api\Data\ItemRequestExtensionInterface|null
     */
    public function getExtensionAttributes(): ?ItemRequestExtensionInterface;

    /**
     * Set an extension attributes object
     *
     * @param \Magento\InventorySourceSelectionApi\Api\Data\ItemRequestExtensionInterface $extensionAttributes
     * @return void
     */
    public function setExtensionAttributes(
        \Magento\InventorySourceSelectionApi\Api\Data\ItemRequestExtensionInterface $extensionAttributes
    ): void;
}
```

Each item contains only SKU and Quantity. 
In the default implementation we don't provide any additional data, such as `Shipping address`, because Priority based algorithm no need this data - just the priority of Sources assigned to given Stock matters. 

For more sophisticated algorithms, like a distance-based additional input data may be needed (i.e. Shipping Address), that's why `InventoryRequestInterface` implements ExtensibleDataInterface interfaces and could be extended with these custom input parameters